archive_verify/workers.py

```diff
@@ -93,48 +30,62 @@ def compare_md5sum(archive_dir):
     else: 
         return True
 
-def verify_archive(archive, archive_path, description, config):
+
+def get_pdc_client_class(config):
     """
-    Our main worker function. This will be put into the RQ/Redis queue when the /verify endpoint gets called. 
-    Downloads the specified archive from PDC and then verifies the MD5 sums. 
+    Determines which PDC Client should be used.
 
-    :param archive: The name of the archive we shall download
-    :param archive_path: The path to the archive on PDC
-    :param description: The unique description that was used when uploading the archive to PDC
     :param config: A dict containing the apps configuration
-    :returns A JSON with the result that will be kept in the Redis queue
+    :returns A PDC Client.
     """
-    dest_root = config["verify_root_dir"]
-    dsmc_log_dir = config["dsmc_log_dir"]
-    whitelist = config["whitelisted_warnings"]
+    return MockPdcClient if config.get("pdc_client", "PdcClient") == "MockPdcClient" else PdcClient
+
 
+def configure_log(dsmc_log_dir, archive_pdc_description):
     now_str = datetime.datetime.now().strftime('%Y-%m-%d_%H-%M')
     log.setLevel(logging.DEBUG)
-    fh = logging.FileHandler(os.path.join(dsmc_log_dir, "{}-{}.log".format(description, now_str)))
+    fh = logging.FileHandler(os.path.join(dsmc_log_dir, "{}-{}.log".format(archive_pdc_description, now_str)))
     fh.setLevel(logging.DEBUG)
     formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
     fh.setFormatter(formatter)
     log.addHandler(fh)
 
-    log.debug("verify_archive started for {}".format(archive))
 
-    job_id = rq.get_current_job().id
-    dest = "{}_{}".format(os.path.join(dest_root, archive), job_id)
+def verify_archive(archive_name, archive_pdc_path, archive_pdc_description, config):
+    """
+    Our main worker function. This will be put into the RQ/Redis queue when the /verify endpoint gets called. 
+    Downloads the specified archive from PDC and then verifies the MD5 sums. 
 
-    download_ok = download_from_pdc(archive_path, description, dest, dsmc_log_dir, whitelist)
+    :param archive_name: The name of the archive we shall download
+    :param archive_pdc_path: The path in PDC TSM to the archive that we want to download
+    :param archive_pdc_description: The unique description that was used when uploading the archive to PDC
+    :param config: A dict containing the apps configuration
+    :returns A JSON with the result that will be kept in the Redis queue
+    """
+    dsmc_log_dir = config["dsmc_log_dir"]
+    configure_log(dsmc_log_dir, archive_pdc_description)
+    log.debug("verify_archive started for {}".format(archive_name))
+
+    pdc_class = get_pdc_client_class(config)
```

Comment:
I would typically call this type of construct `pdc_client_factory` or something similar, but that might just be to much time in Java speaking. I think it makes the intention somewhat clearer.

Reply:
I was struggling with the naming here, good suggestion!